### PR TITLE
feat: storing accounts locally after authentication

### LINKED ISSUE #3: Implement storing accounts locally after authentication

### PR DESCRIPTION
**Pull request by IgorShadurin**
close #3

### CODE REVIEW

**Comment by nugaon** [comment]
please rebase the branch to the current master

File: src/command/account/login.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    type: 'string',
+    required: { when: 'quiet' },
+    minimumLength: MIN_PASSWORD_LENGTH,
+    maximumLength: MAX_PASSWORD_LENGTH,
+  })
+  public portablePassword!: string
+
+  public async run(): Promise<void> {
+    await super.init()
+    this.validateBeeDebugAvailable()
```
Comment (nugaon) [inline src/command/account/login.ts:43]: why is it necessary? read operations are required only with bee for login.

File: src/service/account/index.ts [hunk 2]
```diff
@@ -3,15 +3,71 @@ import { CommandLog } from '../../command/root-command/command-log'
...
+import { ADDRESS_LENGTH, v3ToMnemonic } from '../../utils/wallet'
+import { EncryptedSeed } from './types/wallet'
+
+/**
+ * Available account types
+ */
+export enum AccountType {
+  mnemonic = 'mnemonic',
+  seed = 'seed',
+  all = 'all',
```
[the remark this answers is not in the document]
Comment (nugaon) [reply src/service/account/index.ts:15]: now I see, you are using this enum for sorting account types, so to speak these are query params, aren't they?
Please provide a more verbose JSDoc about the intention how to use this enum.

**Review by nugaon — COMMENTED** [review]
I was thinking, maybe would that be a possible solution to generate a V3 compatible wallet immediately for the saved wallet at login? I think it is possible and then the portable account is even more portable because it can be imported without transformation and you handle lesser wallet types.

File: src/command/account/login.ts (hunk 1 above)
Comment (IgorShadurin) [reply src/command/account/login.ts:43]: This is needed to initialize fdp-storage. With the new version, this is no longer necessary.

**Comment by IgorShadurin** [comment]
If you create a wallet from a seed using ethers, then this wallet does not contain a seed. 
Ethers code: `utils.HDNode.fromSeed(seed)`
Such a wallet contains only the `privateKey, publicKey, parentFingerprint, fingerprint, address, chainCode, index, depth` fields.
Therefore, storing such a wallet, we still need to store the seed somewhere. And by storing only the seed, a wallet can be generated from it on the fly (but with the help it will not be possible to manage the account data).

File: src/utils/encryption.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  })
+
+  return iv.concat(cipherParams.ciphertext)
+}
+
+/**
+ * Encrypt seed with password
+ */
+export function encryptSeed(seed: Uint8Array, password: string): string {
+  const encryptedBytes = encryptBytes(password, bytesToWordArray(seed))
```
[the remark this answers is not in the document]
Comment (IgorShadurin) [reply src/utils/encryption.ts:56]: Pad data is not added for local storing. They are only added in fdp-storage before the portable account is uploaded.

File: src/command/account/login.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    description: 'Username for the portable FDS account',
+    type: 'string',
+    required: true,
+    minimumLength: MIN_USERNAME_LENGTH,
+    maximumLength: MAX_USERNAME_LENGTH,
+  })
+  public username!: string
+
+  @Option({
+    key: 'portable-password',
```
Comment (nugaon) [inline src/command/account/login.ts:33]: maybe would be nice to add alias like `-p`

File: src/command/account/login.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import { assertBytes } from '../../utils/types'
+import { Utils } from '@ethersphere/bee-js'
+
+export class Login extends AccountCommand implements LeafCommand {
+  public readonly name = 'login'
+
+  public readonly description = 'Login to a portable FDS account'
+
+  @Argument({
+    key: 'username',
```
Comment (nugaon) [inline src/command/account/login.ts:23]: alis with  `-u`?

File: src/command/root-command/index.ts [hunk 6]
```diff
@@ -95,7 +99,9 @@ export class RootCommand {
       this.maybeSetFromConfig(option)
     })
 
-    this.fdpStorage = new FdpStorage(this.beeApiUrl, this.beeDebugApiUrl)
+    const batchId = this.beePostageBatchId || (await getUsableBatch())
```
Comment (nugaon) [inline src/command/root-command/index.ts:102]: the problem with this is not all commands require valid postage stamp as some do not upload anything.
either introduce a property in the rootCommand which is a flag for `postageStampRequired` or define postage stamp option on upload commands, but like this is not correctly defined.

**Review by nugaon — APPROVED** [review]
Please correct the postage stamp option definition and then it can be merged

**Pull request merged**